### PR TITLE
helm: enable custom initcontainers

### LINKED ISSUE #4469: Update helm chart to support Downloading GeoIP databases
### Background

[v0.35.0-rc.0](https://github.com/grafana/agent/releases/tag/v0.35.0-rc.0) adds support for `stage.geoip` but https://github.com/oschwald/maxminddb-golang only support loading the GeoIP database from a local path

```
stage.geoip {
  source  = "ip"
  db      = "/path/to/db/GeoLite2-City.mmdb"
  db_type = "city"
}
```

The current helm chart does not provide anyway to enable the database files to be added to the pod file system when created. 

### Proposal

A solution to the problem would be to add https://hub.docker.com/r/maxmindinc/geoipupdate as an optional sidecar that can be responsible for ensuring that the required GeoIP files are available in the pod.

### PR DESCRIPTION
**Pull request by dwalker-sabiogroup**
#### PR Description

Release 0.35.0 adds GeoIP processing for logs but provides no mechanism for downloading the required database files to the agent pod. This change enables integration with ghcr.io/maxmind/geoipupdate:v6.0 which is the recommended method to download and update GeoIP database files

#### Which issue(s) this PR fixes

Fixes #4469 

#### Notes to the Reviewer

This is a resubmit of https://github.com/grafana/agent/pull/4486

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Main (unreleased)
 
 - Add [godeltaprof](https://github.com/grafana/godeltaprof) profiling types (`godeltaprof_memory`, `godeltaprof_mutex`, `godeltaprof_block`) to `pyroscope.scrape` component
 - Integrations: make `udev` data path configurable in the `node_exporter` integration. (@sduranc)
+- Add support for initcontainers in helm chart. (@dwalker-sabiogroup)
```
Comment (tpaschalis) [inline CHANGELOG.md:17]: Could we please move this entry to the Helm chart's CHANGELOG instead of here?

**Review by tpaschalis — COMMENTED** [review]
Just a couple of small nits, it's looking nice!

**Comment by dwalker-sabiogroup** [comment]
@tpaschalis - I have fixed up the couple of nits.  Can you please review.

**Review by tpaschalis — APPROVED** [review]
LGTM, thank you!

**Pull request merged**